docs/proposals/prevent-removal-managed-resources/README.md

```diff
@@ -0,0 +1,63 @@
+---
+title: Prevent removal managed resources
+authors:
+- "@CharlesQQ"
+reviewers:
+- "@RainbowMango"
+- "@XiShanYongYe-Chang"
+- "@chaunceyjiang"
+approvers:
+
+creation-date: 2024-06-27
+
+# prevent removal managed resource
+
+## Summary
+
+A deletion mechanism and its API is provided to prevent cascade resource in member cluster deleted by control plane.
+
+## Motivation
+
+All resources reconciled by the execution controller have a finalizer `karmada.io/execution-controller` added to work object's metedata.
+This finalizer will prevent deletion of work object until the execution controller has a chance to perform pre-deletion cleanup.
+
+However, in some cases, user want to delete resource in control plane but resource in member cluster kept, like `kubectl delete sts xxx --cascade=orphan`, sts is deleted but pods not.
+Karmada have different types of resources, different resource might have different prevent removal process.
+
+1. resource template: clean up the labels/annotations from the resource in member clusters and delete work object.
+2. dependent resource: clean up the labels/annotations from the dependent resource in member clusters,such as configmap/secret,  and then delete work object.
```

Comment:
<img width="879" alt="image" src="https://github.com/karmada-io/karmada/assets/17962021/12d81b5f-219d-4be0-b6af-e3337996a194">
When we delete the `deploymentA` of the control plane, do we support retaining the `cm` in `member1` cluster, but deleting the `cm` in `member2` cluster?

Reply:
I don't think so, at least for now based on the use cases we collected.

By the way, a little confused by the `deploymentB` in the picture, are you trying to say the case that a configmap shared with multiple deployments?

Reply:
> a configmap shared with multiple deployments?

Yes.

This image mainly wants to express two things:
1. Can we control whether dependent resources are kept separately in the member clusters?
2. Can we control whether resources are kept in specific member clusters?